jest/jest-tests.ts

```diff
@@ -367,3 +425,53 @@ describe('strictNullChecks', function () {
         done();
     })
 });
+
+class TestApi {
+    constructor() { };
+    testProp: boolean;
+    testMethod(a: number): string { return ""; }
+}
+
+declare function mockedFunc(a: number): string;
+
+describe('Mocked type', function () {
+    it('Works', function () {
+        const mock: jest.Mocked<TestApi> = new TestApi() as any;
+        mock.testProp;
+        mock.testMethod.mockImplementation(() => 'test');
+        mock.testMethod(5).toUpperCase();
+    });
+});
+
+describe('Mocks', function () {
+    it('jest.fn() without args is a function type', function () {
+        const test = jest.fn();
+        test();
+        new test();
+        test.mock.instances[0];
+        test.mockImplementation(() => { });
+    });
+
+    it('jest.fn() with returned object infers type', function () {
+        const testMock = jest.fn(() => ({ a: 5, test: jest.fn() }));
+
+        testMock(5, 5, 'a');
+        testMock.mockImplementation(() => { });
+        testMock.caller;
+
+        const ins = new testMock();
```

Comment:
I think no. Just calling mock function, rather than instantiating it with new, will return any type. I think it's fine since you usually don't need to call mock functions in your tests to get the value - they're are being passed into SUT in most common case. Constructor case is different since you can use Partial<MyAPI> type which will be inferred into full MyAPI type and you can pass it under SUT without compiler errors and casting it to any.